src/Microsoft.TemplateEngine.Cli/TemplateResolution/TemplateResolver.cs

```diff
@@ -485,5 +443,197 @@ private static void AddParameterMatchingToTemplates(IReadOnlyCollection<ITemplat
                 }
             }
         }
+
+        /// <summary>
+        /// The method makes all short names in template group available for matching for given template (using <see cref="TemplateInfoWithGroupShortNames"/> <see cref="ITemplateInfo"/> implementation).
+        /// TODO: double-check if that is needed. If needed that should be replaced by TemplateGroup logic as:
+        /// - group templates by template groups
+        /// - match templates through filters by template group including this logic of sharing short names
+        /// - create resolution result afterwards.
+        /// </summary>
+        /// <param name="templateList">the list of <see cref="ITemplateInfo"/> to process.</param>
+        /// <returns></returns>
+        private static IReadOnlyList<ITemplateInfo> SetupFilterableTemplateInfoFromTemplateInfo(IReadOnlyList<ITemplateInfo> templateList)
+        {
+            Dictionary<string, HashSet<string>> shortNamesByGroup = new Dictionary<string, HashSet<string>>();
+
+            // get the short names lists for the groups
+            foreach (ITemplateInfo template in templateList)
+            {
+                string effectiveGroupIdentity = !string.IsNullOrEmpty(template.GroupIdentity)
+                    ? template.GroupIdentity
+                    : template.Identity;
+
+                if (!shortNamesByGroup.TryGetValue(effectiveGroupIdentity, out HashSet<string> shortNames))
+                {
+                    shortNames = new HashSet<string>();
+                    shortNamesByGroup[effectiveGroupIdentity] = shortNames;
+                }
+
+                if (template is IShortNameList templateWithShortNameList)
```

Comment:
We are marking this absolute in another PR. Are we going to remove this section later?

Reply:
yep, depending on what is merged first